src/charm.py

```diff
@@ -303,6 +302,19 @@ def logs_rules(self) -> Dict[str, Any]:
         """Return a list of logging rules."""
         return self._cos.logs_alerts
 
+    @property
+    def requested_receivers(self) -> set:
```

Comment:
the naming in this method could be improved.
We have 
- 'requested receivers' (function name) 
- 'requested protocols' (wrapped method on `self._cos`) 
- 'protocols' (return object)

Given that not all readers of the gagent codebase will be familiar with what a 'receiver' is, who might be requesting them, etc... I say we hide both from the public api and avoid the word 'receiver' as much as possible.

How about:

```python
def _requested_tracing_protocols ...:
    requested_protocols = self._cos.requested_protocols()
    ...
    return requested_protocols
```

Reply:
Not sure if I like the underscore before (it would make it different from `metrics_rules`, `job_rules` and other methods above that also expose the content passed from the charm using cos-agent, but I agree on the protocols part.

On the other side of the databag I think we still need to return `receivers` as `protocols` won't match the returned data. Unless you have a better idea :)